test/common/http/conn_manager_impl_test.cc

```diff
@@ -897,6 +899,56 @@ TEST_F(HttpConnectionManagerImplTest, RouteShouldUseSantizedPath) {
   conn_manager_->onData(fake_input, false);
 }
 
+TEST_F(HttpConnectionManagerImplTest, PreseveUpstreamDateDisabledDateNotSet) {
+  setup(false, "");
+  setUpEncoderAndDecoder(false, false);
+  sendRequestHeadersAndData();
+  preserve_upstream_date_ = false;
+  const auto* modifiedHeaders = sendResponseHeaders(
```

Comment:
nit: s/modifiedHeaders/modified_headers here and below (I think we can get clang-tidy to check this at some point)